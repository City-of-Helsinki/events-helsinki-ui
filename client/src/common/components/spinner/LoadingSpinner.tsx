import React, { FunctionComponent } from "react";

import styles from "./loadingSpinner.module.scss";

const LoadingSpinner: FunctionComponent<{ isLoading: boolean }> = ({
  isLoading,
  children
}) => {
  return (
    <>
      {isLoading ? (
        <div className={styles.spinnerWrapper}>
          <div className={styles.spinner}>
            <div />
          </div>
        </div>
      ) : (
        children
      )}
    </>
  );
};

export default LoadingSpinner;
