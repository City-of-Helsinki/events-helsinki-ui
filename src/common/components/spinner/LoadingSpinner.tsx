import classNames from "classnames";
import React, { FunctionComponent } from "react";

import styles from "./loadingSpinner.module.scss";

interface Props {
  hasPadding?: boolean;
  isLoading: boolean;
}

const LoadingSpinner: FunctionComponent<Props> = ({
  hasPadding = true,
  isLoading,
  children
}) => {
  return (
    <>
      {isLoading ? (
        <div
          className={classNames(styles.spinnerWrapper, {
            [styles.hasPadding]: hasPadding
          })}
        >
          <div className={styles.spinner}>
            <div />
          </div>
        </div>
      ) : (
        children
      )}
    </>
  );
};

export default LoadingSpinner;
