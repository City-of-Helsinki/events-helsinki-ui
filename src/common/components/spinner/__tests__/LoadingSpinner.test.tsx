import { shallow } from 'enzyme';
import React from 'react';

import LoadingSpinner from '../LoadingSpinner';

it('LoadingSpinner matches snapshot', () => {
  const spinner = shallow(<LoadingSpinner isLoading={true} />);
  expect(spinner.html()).toMatchSnapshot();
});

it('render spinner if isLoading is true', () => {
  const spinner = shallow(<LoadingSpinner isLoading={true} />).children();
  expect(spinner.prop('className')).toContain('spinner');
});

it('render child component if isLoading is false', () => {
  const spinner = shallow(
    <LoadingSpinner isLoading={false}>
      <div className="component"></div>
    </LoadingSpinner>
  ).children();
  expect(spinner.prop('className')).toEqual('component');
});
