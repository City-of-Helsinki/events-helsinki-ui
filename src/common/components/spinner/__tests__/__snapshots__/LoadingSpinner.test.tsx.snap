// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="spinnerWrapper hasPadding"
>
  <div
    class="spinner"
  >
    <svg
      class=""
      height="100%"
      viewBox="0 0 70 40"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill-rule="evenodd"
        stroke-width="1"
      >
        <g
          transform="translate(-243.000000, -825.000000)"
        >
          <path
            d="M278,835.434783 C264.604333,835.434783 253.719637,846.118748 253.503282,859.379973 L253.5,859.782609 C253.5,862.664094 251.149495,865 248.25,865 C245.350505,865 243,862.664094 243,859.782609 L243,859.782609 C243,840.572704 258.670034,825 278,825 C297.136667,825 312.686233,840.262808 312.995311,859.207415 L313,859.782609 C313,862.664094 310.649495,865 307.75,865 C304.850505,865 302.5,862.664094 302.5,859.782609 L302.5,859.782609 C302.5,846.335676 291.530976,835.434783 278,835.434783 Z"
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;
